tracing-attributes/src/lib.rs

```diff
@@ -210,9 +210,18 @@ pub fn instrument(args: TokenStream, item: TokenStream) -> TokenStream {
             FnArg::Typed(PatType { pat, .. }) => param_names(*pat),
             FnArg::Receiver(_) => Box::new(iter::once(Ident::new("self", param.span()))),
         })
-        .filter(|ident| !skips.contains(ident))
         .collect();
 
+    for skip in &skips {
+        if !param_names.contains(skip) {
+            return quote_spanned!(skip.span()=>
+                compile_error!("attempting to skip non-existing parameter")
```

Comment:
nit: i think this should be 
```suggestion
                compile_error!("attempting to skip non-existent parameter")
```